cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdminTest.java

```diff
@@ -125,13 +125,11 @@ public void testConfigUpdateFailures() throws Exception {
     Assert.assertTrue(namespaceAdmin.exists(namespaceId));
 
     // Updating the root directory for a namespace should fail
-    try {
-      namespaceAdmin.updateProperties(nsMeta.getNamespaceId().toId(),
-                                      new NamespaceMeta.Builder(nsMeta).setRootDirectory("/newloc").build());
-      Assert.fail();
-    } catch (BadRequestException e) {
-      // expected
-    }
+    namespaceAdmin.updateProperties(nsMeta.getNamespaceId().toId(),
+                                    new NamespaceMeta.Builder(nsMeta).setRootDirectory("/newloc").build());
+
+    // removing the root directory mapping for a namespace should fail
+    namespaceAdmin.updateProperties(nsMeta.getNamespaceId().toId(), new NamespaceMeta.Builder(nsMeta).build());
```

Comment:
this won't work, since the test will exit at line 128. If you want to test multiple such scenarios, then you should ``try...catch`` here.